pkg/provider/gitlab/gitlab.go

```diff
@@ -214,26 +267,53 @@ func (g *Gitlab) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretData
 	// 	"masked": true,
 	// 	"environment_scope": "*"
 	// }
-
 	var vopts *gitlab.GetProjectVariableOptions
 	if g.environment != "" {
 		vopts = &gitlab.GetProjectVariableOptions{Filter: &gitlab.VariableFilter{EnvironmentScope: g.environment}}
 	}
-	data, _, err := g.client.GetVariable(g.projectID, ref.Key, vopts)
+
+	data, resp, err := g.projectVariablesClient.GetVariable(g.projectID, ref.Key, vopts)
+	if resp.StatusCode >= 400 && resp.StatusCode != 404 && err != nil {
+		return nil, err
+	}
+
+	err = g.ResolveGroupIds()
 	if err != nil {
 		return nil, err
 	}
 
+	var result []byte
+	result, err = extractVariable(ref, data.Value)
+	for i := len(g.groupIDs) - 1; i >= 0; i-- {
+		groupID := g.groupIDs[i]
+		if result != nil {
+			return result, nil
+		}
+
+		groupVar, resp, err := g.groupVariablesClient.GetVariable(groupID, ref.Key, nil)
+		if resp.StatusCode >= 400 && resp.StatusCode != 404 && err != nil {
+			return nil, err
+		}
+		result, _ = extractVariable(ref, groupVar.Value)
```

Comment:
`groupVar` is nil if 404 was returned from `GetVariable()`

Reply:
fixed